File: 204.cpp
// Count the number of prime numbers less than a non-negative number, n.
//
// Example:
//
// Input: 10
// Output: 4
// Explanation: There are 4 prime numbers less than 10, they are 2, 3, 5, 7.

class Solution {
public:
    int countPrimes(int n) {
        vector<bool> is_prime(n, true);
        int res = 0;
        for (int i = 2; i < n; ++i) {
            if (is_prime[i]) {
                res++;
            }
            for (int j = 2; j * i < n; ++j) {
                is_prime[j * i] = false;
            }
        }

        return res;
    }
};

File: Python/204.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 2:
            return 0
        prime = [True]*n
        prime[0] = prime[1] = False
        for i in range(2, n):
            if prime[i]:
                j = 2
                while i*j<n:
                    prime[i*j] = False
                    j+=1
        return prime.count(True)
